drivers/sensor/tmp116/tmp116.c

```diff
@@ -36,31 +36,42 @@ static int tmp116_reg_read(const struct device *dev, uint8_t reg,
 	return 0;
 }
 
+static int tmp116_reg_write(const struct device *dev, uint8_t reg,
+			    uint16_t val)
+{
+	struct tmp116_data *drv_data = dev->data;
+	const struct tmp116_dev_config *cfg = dev->config;
+	uint8_t tx_buf[3] = {reg, val >> 8, val & 0xFF};
+
+	return i2c_write(drv_data->i2c, tx_buf, sizeof(tx_buf),
+			cfg->i2c_addr);
+}
+
 /**
  * @brief Check the Device ID
  *
  * @param[in]   dev     Pointer to the device structure
  *
- * @retval 0 On success
+ * @retval device id On success
  * @retval -EIO Otherwise
  */
-static inline int tmp116_device_id_check(const struct device *dev)
+static inline uint16_t tmp116_device_id_check(const struct device *dev)
```

Comment:
I think it would be fine to keep this as an `int`. Generally, in Zephyr, a negative value is used for errors. Since the `int` is also capable of representing a `uint16_t`, that is covered.

Reply:
Sure, I will change it.